enclave/core/sgx/globals.c

```diff
@@ -8,6 +8,14 @@
 
 extern volatile const oe_sgx_enclave_properties_t oe_enclave_properties_sgx;
 
+/* The memory properties(only the properties, not the actual amount of memory
+   pages etc), are patched during enclave initialization to avoid patches to the
+   base image, which would not be verifiable without having an actual copy of
+   the base image. This means that we need a mutable version of
+   oe_enclave_properties_sgx, which is initialized to a copy thereof, so that
+   non-EEID enclaves won't see a difference. */
+oe_sgx_enclave_properties_t oe_enclave_properties_sgx_mutable;
```

Comment:
I realized that this might not be necessary given the changes in #3024 (which conflicts with this). The use of the `oe_enclave_properties_sgx` in most of the global functions was only for MSVC builds of the enclave, which is currently unsupported anyway. This basically means that everything simplifies down to having to fix 
`__oe_get_heap_size()`. This modifies an earlier suggestion on the implementation of `_eeid_patch_memory_sizes()`: 
* Instead of writing a mutable version of `oe_enclave_properties_sgx` which is mostly unused, make `_have_eeid` a pointer to the EEID instead:
    ```c
    #ifdef OE_WITH_EXPERIMENTAL_EEID
    volatile oe_eeid_t* oe_eeid = NULL;
    #endif
    ```
* In `_eeid_patch_memory_sizes()`, it just sets  `oe_eeid` to the heap provided value.
  ```c
  size_t __oe_get_heap_size()
  {
      #ifdef OE_WITH_EXPERIMENTAL_EEID
      if (oe_eeid)
          return oe_eeid->size_settings.num_heap_pages * OE_PAGE_SIZE;
      else
      #endif
          return oe_enclave_properties_sgx.header.size_settings.num_heap_pages * OE_PAGE_SIZE;
  }

  const void* __oe_get_eeid()
  {
      return oe_eeid;
  }
  ``` 
The TCS count and stack page count isn't referenced anywhere else in the enclave today, but they would similarly need to be accessed through accessor functions in the same way.

Reply:
Awesome, thanks for that, this should simplify that part significantly!

Reply:
Done! This simplified the initialization sequence quite a bit. I did find that I need to save the extended enclave size (which is usually read from the image info), see `__oe_get_enclave_size()` as some checks fail without that. We could re-calculate that in `_eeid_patch_memory_sizes`, but I figured it's less error-prone to just save it next to the offset. Also, I got rid of the magic marker altogether. If we recognize base images by their memory settings, we can just as well do the same during initialization.